frog/imports/ui/StudentView/SessionBody.jsx

```diff
@@ -16,20 +18,74 @@ const getInitialState = (activities, d = 1) => {
       };
 };
 
-const SessionBody = ({ session }: { session: Object }) => {
+const SessionBody = ({
+  session,
+  currentTime
+}: {
+  session: Object,
+  currentTime: number
+}) => {
+  const tmp =
+    session.countdownStartTime > 0
+      ? Math.round(
+          session.countdownStartTime + session.countdownTimeLeft - currentTime
+        )
+      : session.countdownTimeLeft;
+
   if (!session.openActivities || session.openActivities.length === 0) {
-    return <h1>NO ACTIVITY</h1>;
+    return (
+      <div>
+        {session.countdownStartTime !== -1 &&
+          <Countdown>
+            <h4>
+              {msToString(tmp)}
+            </h4>
+          </Countdown>}
+        <h1>NO ACTIVITY</h1>
+      </div>
+    );
```

Comment:
Make this 
```
  <Countdown>
 +            <h4>
 +              {msToString(tmp)}
 +            </h4>
 +          </Countdown
```

into a component, so you don't have to duplicate the code twice.

Reply:
actually thrice ;)